feature_engine/creation/math_features.py

```diff
@@ -202,14 +265,117 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         X_new: Pandas dataframe, shape = [n_samples, n_features + n_operations]
             The input dataframe plus the new variables.
         """
+
+        def np_transform(np_df, new_variable_names, np_variables, np_functions):
+            np_result_df = pd.DataFrame()
+            for np_function_idx, np_function in enumerate(np_functions):
+                if callable(np_function):
```

Comment:
if it is a callable, can we not apply it straightaway? is not getting the name and then the if loop adding complexity?